### PR TITLE
Add docs for how to merge modules.json

### PR DESCRIPTION
**Pull request by airportyh**
## Why?

I've set up a strategy for dealing with conflicts in `modules.json`

## Changes

Added instructions.

### CODE REVIEW

**Review by ryantm — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -158,3 +158,48 @@ or see it without building:
...
+This is a race condition! For this reason, hand-merging `modules.json` is a bad idea.
+
+We have CI checks in place to ensure:
+
+1. you cannot merge a PR if it's not in sync with `main`
+2. the commits referenced in `modules.json` all exist in the linear history of your branch
+
+When you have a conflict in `modules.json` you should:
+
+1. take the version from upstream whole
```
[the remark this answers is not in the document]
Comment (airportyh) [reply README.md:175]: is that wording confusing. I mean take all of its contents whole.

[the remark this answers is not in the document]
Comment (airportyh) [reply README.md:175]: "take the version from upstream in its entirety"

File: README.md [hunk 2]
```diff
@@ -158,3 +158,49 @@ or see it without building:
...
+3. re-run `python scripts/lock_modules.py`
+4. make another commit
+
+To do the above you can either rebase or merge.
+
+With the rebase approach (pros: do without a merge commit):
+```
+git fetch
+git rebase origin/main
+git checkout --ours modules.json
```
Comment (cdmistman) [inline README.md:186]: Shouldn't this be theirs? Otherwise we're erasing previous commits right?

**Review by cdmistman — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (airportyh) [reply README.md:186]: when rebasing it's reversed

**Review by cdmistman — APPROVED** [review]

**Pull request merged**